R/oc_check_query.R

```diff
@@ -0,0 +1,192 @@
+#' Check OpenCage query arguments
+#'
+#' Function that checks the query arguments passed to OpenCage
+#'
+#' @param limit The maximum number of results that should be returned. Integer
+#'   values between 1 and 100 are allowed.
+#' @inheritParams oc_forward
+#'
+#' @keywords internal
+
+
+oc_check_query <-
+  function(
+    placename = NULL,
+    latitude = NULL,
+    longitude = NULL,
+    key = NULL,
+    bounds = NULL,
+    countrycode = NULL,
+    language = NULL,
+    limit = NULL,
+    min_confidence = NULL,
+    no_annotations = NULL,
+    no_dedupe = NULL,
+    no_record = NULL,
+    abbrv = NULL,
+    add_request = NULL
+  ) {
+    arglist <-
+      purrr::compact(
+        list(
+          placename = placename,
+          latitude = latitude,
+          longitude = longitude,
+          bounds = bounds,
+          countrycode = countrycode,
+          language = language,
+          limit = limit,
+          min_confidence = min_confidence,
+          no_annotations = no_annotations,
+          no_dedupe = no_dedupe,
+          abbrv = abbrv,
+          add_request = add_request
+        )
+      )
+
+    # prevent obscure warning message from pwalk if length(arglist) == 0
+    stopifnot(length(arglist) >= 1)
+
+    purrr::pwalk(
+      .l = arglist,
+      .f = .oc_check_query,
+      key = key,
+      no_record = no_record
+    )
+  }
+
+.oc_check_query <-
+  function(
+    placename = NULL,
+    latitude = NULL,
+    longitude = NULL,
+    key = NULL,
+    bounds = NULL,
+    countrycode = NULL,
+    language = NULL,
+    limit = NULL,
+    min_confidence = NULL,
+    no_annotations = NULL,
+    no_dedupe = NULL,
+    no_record = NULL,
+    abbrv = NULL,
+    add_request = NULL
+  ) {
+    # check placename
+    if (!is.null(placename)) {
+      if (!is.character(placename)) {
+        stop(call. = FALSE, "`placename` must be a character vector.")
+      }
+    }
+
+    # check latitude
+    if (!is.null(latitude)) {
+      if (!(dplyr::between(latitude, -90, 90))) {
+        stop(call. = FALSE, "`latitude` must be between -90 and 90.")
+      }
+    }
+
+    # check longitude
+    if (!is.null(longitude)) {
+      if (!(dplyr::between(longitude, -180, 180))) {
+        stop(call. = FALSE, "`longitude` must be between -180 and 180.")
+      }
+    }
+
+    # check key
+    if (is.null(key)) {
+      stop(call. = FALSE, "A `key` must be provided.")
+    } else if (!is.character(key)) {
+      stop(call. = FALSE, "`key` must be a character vector.")
+    } else if (length(key) > 1) {
+      stop(call. = FALSE, "`key` must be a vector of length one.")
+    }
+
+    # check bounds
+    if (!is.null(bounds)) {
+      if (length(bounds) != 4) {
+        stop(
+          call. = FALSE,
+          "Every `bbox` must be a numeric vector of length 4."
```

Comment:
I see here you write "Every", should you write "Every xmin" in oc_check_bbox too?

Reply:
Well spotted, yes.